pkg/clustermanager/eksa_installer.go

```diff
@@ -108,6 +109,47 @@ func (i *EKSAInstaller) Upgrade(ctx context.Context, log logr.Logger, c *types.C
 	return changeDiff, nil
 }
 
+// ApplyBundles applies the bundles to the cluster.
+func (i *EKSAInstaller) ApplyBundles(ctx context.Context, log logr.Logger, cluster *types.Cluster, spec *cluster.Spec) error {
+	bundleObj, err := yaml.Marshal(spec.Bundles)
+	if err != nil {
+		return fmt.Errorf("outputting bundle yaml: %v", err)
+	}
+
+	log.V(1).Info("Applying Bundles to cluster")
+	if err := i.client.ApplyKubeSpecFromBytes(ctx, cluster, bundleObj); err != nil {
+		return fmt.Errorf("applying bundle spec: %v", err)
+	}
+
+	return nil
+}
+
+// ApplyReleases applies the releases to the cluster.
+func (i *EKSAInstaller) ApplyReleases(ctx context.Context, log logr.Logger, cluster *types.Cluster, spec *cluster.Spec) error {
+	releaseObj, err := yaml.Marshal(spec.EKSARelease)
+	if err != nil {
+		return fmt.Errorf("outputting release yaml: %v", err)
+	}
+
+	log.V(1).Info("Applying EKSA Release to cluster")
+	if err := i.client.ApplyKubeSpecFromBytes(ctx, cluster, releaseObj); err != nil {
+		return fmt.Errorf("applying release spec: %v", err)
+	}
+
+	return nil
+}
+
+// CreateNamespaceIfNotPresent creates the cluster namespace if it doesn not exist.
+func (i *EKSAInstaller) CreateNamespaceIfNotPresent(ctx context.Context, log logr.Logger, cluster *types.Cluster, spec *cluster.Spec) error {
+	if spec.Cluster.Namespace != "" {
+		if err := i.client.CreateNamespaceIfNotPresent(ctx, cluster.KubeconfigFile, spec.Cluster.Namespace); err != nil {
```

Comment:
IMO we should not do this
It should be the use the one creating the namespace before calling eks-a if they want to create a cluster in it